### DISCUSSION TITLE
Chapter 7: Foundry in Docker. DevOpsTools returns No such file or directory

### DISCUSSION THREAD

**Discussion by shobhit-gupta** [Cyfrin/foundry-full-course-cu#501] [Q&A] [answered]
Hello,

I've setup foundry in a dockerized environment. `Dockerfile` & `docker-compose.yml` attached below if anyone wants to refer.  Everything works pretty fine except when I try to run the script `Interactions.s.sol` methods from the terminal.

Note: I've already installed `jq`.

Here's the process I followed:
1. `make anvil`: To run a local anvil node.
2. `make deploy`: Using a new terminal, I ran this command. Presumably this deployed the contract on Anvil chain.
3. `make fund`: I receive the error : `[0] VM::ffi([bash, /app/lib/foundry-devops/src/get_recent_deployment.sh, FundMe, 31337, /app//broadcast]) "Failed to execute command: No such file or directory (os error 2)"` 

The path `app//broadcast` mentioned in the error does seem wrong. So I also tried to make some changes to the file I think the error is originating from: `DevOpsTools.sol` in `get_most_recent_deployment` function. But, to no avail.

I would really appreciate some help.

<img width="1681" alt="Screenshot 2023-07-31 at 9 42 48 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/25628519/c7eeff35-b2a9-4925-bcd9-2b5ca719ce61">

<img width="339" alt="Screenshot 2023-07-31 at 9 43 24 PM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/25628519/27710c44-cb4d-4f98-a96a-b1dc0323f5a1">

```Dockerfile
# Use the latest foundry image
FROM ghcr.io/foundry-rs/foundry

# Copy our source code into the container
WORKDIR /app

RUN apk add --no-cache make git jq

COPY . .

# Default port for anvil
EXPOSE 8545
```

```docker-compose.yml
services:
  fund-me:
    build: ./
    stdin_open: true
    tty: true
    container_name: fund-me-app
    ports:
      - "12545:8545"
    volumes:
      - ./:/app
    cpu_count: 6
    mem_limit: 6g
```

**Comment by Niferu** [comment]
Hello @shobhit-gupta 

I see that you have made some changes to `get_recent_deployments.sh`. Check with original devops repository if code is the same compared to yours. You can also link your repo, maybe I will be able to find something else what is wrong.

**Reply by shobhit-gupta** [reply]
Hello @Neftyr 

Thank you for taking out some time to help me! 
Here's the link to the repo: https://github.com/shobhit-gupta/blockchain-foundry-fundme

As you can see I'm using a VSCode Dev Container. So I think, it should work fine without any specific instruction if you open it in VSCode & have the Docker on your system.

Also, I made (and **reverted**) changes to the **function** `get_most_recent_deployment` in **DevOpsTools.sol file**. Specifically around line no. _91_ to get rid of the extra `/` I saw in the error message. Please note, I **didn't** make any changes to the script `get_recent_deployments.sh`.

**Reply by Niferu** [reply, marked as answer]
Hi @shobhit-gupta 

Chekout this topic: [link](https://unix.stackexchange.com/questions/433444/cant-run-script-file-in-docker-no-such-file-or-directory)

**Reply by shobhit-gupta** [reply]
Hi @Neftyr 

You identified the correct issue! `Alpine` images don't come with bash. So, I added `bash` to the install list in the Dockerfile. And now it works! 

Thank you!

**Reply by Niferu** [reply]
Glad you have fixed it and that I could help cheers! :)
